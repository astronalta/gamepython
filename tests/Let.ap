# aptest print_ast on
Let < Object {

    let1() {
        let a Int = 0, b Int {

        } 
    }

}
