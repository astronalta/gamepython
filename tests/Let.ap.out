Module
  name: 
  feature0: Class
    name: Let
    comment:
    feature0: Function
      name: let1
      type: Void
      comment:
      formal0: Formal
        name: self
        type: Self
      block: Block
        child0: Let
          variable0: Assignment
            name: a
            type: Int
            declared: Int
            initializer: IntegerLiteral(0)
          block: Block
    feature1: Function
      name: @destroy
      type: Void
      formal0: Formal
        name: self
        type: Self
      block: Block
    feature2: Function
      name: @init
      type: Void
      block: Block
